### PR TITLE
[yang][smartswitch] Add YANG model for MID_PLANE_BRIDGE and DPU tables.

### PR DESCRIPTION
**Pull request by oleksandrivantsiv**
#### Why I did it
Add the YANG model according to [Smart Switch IP address assignment HDL](https://github.com/sonic-net/SONiC/pull/1504).

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Implement new YANG model containers.

#### How to verify it
Run YANG model unit tests. The changes add new unit tests to cover new functionality.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: src/sonic-yang-models/yang-models/sonic-smart-switch.yang [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+				}
+
+				leaf midplane_interface {
+					description "Name of the interface that represents DPU";
+
+					type string {
+						pattern "dpu[0-9]+";
+					}
+
+					must "(current() = current()/../dpu_name)";
```
Comment (yaqiangz) [inline src/sonic-yang-models/yang-models/sonic-smart-switch.yang:71]: One simple question, this must condition qualifies midplane_interface and dut_name to be equal, so why is this midplane_interface needed?

**Review by dgsudharsan — APPROVED** [review]

File: src/sonic-yang-models/tests/files/sample_config_db.json [hunk 2]
```diff
@@ -2534,6 +2534,20 @@
                 "bank": "1",
                 "link": "PortChannel2"
             }
+        },
+        "MID_PLANE_BRIDGE": {
+            "GLOBAL" : {
+                "bridge": "bridge_midplane",
```
[the remark this answers is not in the document]
Comment (oleksandrivantsiv) [reply src/sonic-yang-models/tests/files/sample_config_db.json:2540]: Correct. We have 1 mid plane bridge and all DPU interfaces are this bridge

File: src/sonic-yang-models/yang-models/sonic-smart-switch.yang (hunk 1 above)
Comment (oleksandrivantsiv) [reply src/sonic-yang-models/yang-models/sonic-smart-switch.yang:71]: `dpu_name` is a table key. For now, we have only one `midplane_interface` filed under the key. In the future, we will have more information there. `midplane_interface` represents the name of the interface in Linux connected to the DPU. To simplify the mapping from the DPU to its interface in Linux we require to use the same name. The entry in the `DHCP_SERVER_IPV4_PORT` table refers to the `midplane_interface` field.

**Comment by liat-grozovik** [comment]
@yaqiangz could you please review following the changes?

**Review by yaqiangz — APPROVED** [review]
LGTM

**Comment by yaqiangz** [comment]
@ganglyu Could you help to review this PR?

File: src/sonic-yang-models/yang-models/sonic-smart-switch.yang [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+				key "dpu_name";
+
+				leaf dpu_name {
+					description "Name of the DPU";
+					type string {
+						pattern "dpu[0-9]+";
+					}
+				}
+
+				leaf midplane_interface {
```
[the remark this answers is not in the document]
Comment (oleksandrivantsiv) [reply src/sonic-yang-models/yang-models/sonic-smart-switch.yang:64]: `dpu_name` is a key of the table and represents the name of the DPU.
`midplane_interface` is the attribute of the DPU and represents the name of the midplane (PCIe) interface.
Even though they share the same name (for convenience) they represent the different entities.

**Comment by oleksandrivantsiv** [comment]
> Would you please update src/sonic-yang-models/doc/Configuration.md?

@ganglyu done

**Review by ganglyu — APPROVED** [review]

**Review by liat-grozovik — APPROVED** [review]

**Pull request merged**